qualtran/bloqs/mod_arithmetic/mod_subtraction_test.py

```diff
@@ -0,0 +1,96 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+import sympy
+
+import qualtran.testing as qlt_testing
+from qualtran import QMontgomeryUInt, QUInt
+from qualtran.bloqs.mod_arithmetic.mod_subtraction import _cmod_neg, _mod_neg, CModNeg, ModNeg
+from qualtran.resource_counting import GateCounts, QECGatesCost, query_costs
+from qualtran.resource_counting.generalizers import ignore_alloc_free, ignore_split_join
+
+
+@pytest.mark.parametrize('dtype', [QUInt, QMontgomeryUInt])
+@pytest.mark.parametrize(
+    ['bitsize', 'prime'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length(), 10)]
+)
+def test_valid_modneg_decomposition(dtype, bitsize, prime):
+    b = ModNeg(dtype(bitsize), prime)
+    qlt_testing.assert_valid_bloq_decomposition(b)
+    qlt_testing.assert_equivalent_bloq_counts(b, [ignore_split_join, ignore_alloc_free])
+
+
+@pytest.mark.parametrize('cv', range(2))
+@pytest.mark.parametrize('dtype', [QUInt, QMontgomeryUInt])
+@pytest.mark.parametrize(
+    ['bitsize', 'prime'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length(), 10)]
+)
+def test_valid_cmodneg_decomposition(dtype, bitsize, prime, cv):
+    b = CModNeg(dtype(bitsize), prime, cv)
+    qlt_testing.assert_valid_bloq_decomposition(b)
+    qlt_testing.assert_equivalent_bloq_counts(b, [ignore_split_join, ignore_alloc_free])
+
+
+@pytest.mark.parametrize('dtype', [QUInt, QMontgomeryUInt])
+@pytest.mark.parametrize(
+    ['bitsize', 'prime'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length(), 10)]
+)
+def test_modneg_classical_action(dtype, bitsize, prime):
+    b = ModNeg(dtype(bitsize), prime)
+    cb = b.decompose_bloq()
+    for x in range(prime):
+        assert b.call_classically(x=x) == cb.call_classically(x=x) == ((-x) % prime,)
+
+
+@pytest.mark.parametrize('cv', range(2))
+@pytest.mark.parametrize('dtype', [QUInt, QMontgomeryUInt])
+@pytest.mark.parametrize(
+    ['bitsize', 'prime'], [(p, n) for p in (13, 17, 23) for n in range(p.bit_length(), 10)]
+)
+def test_cmodneg_classical_action(dtype, bitsize, prime, cv):
```

Comment:
mark slow, make (two) quick checks that randomly sample a couple configurations

Reply:
done